Sources/PubNub/Helpers/Crypto/Cryptors/LegacyCryptor.swift

```diff
@@ -0,0 +1,184 @@
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import Foundation
+import CommonCrypto
+
+/// Provides backward-compatible way of encryption/decryption that matches
+/// deprecated [Crypto](https://github.com/pubnub/swift/blob/master/Sources/PubNub/Helpers/Crypto/Crypto.swift#L32)
+///
+/// - Important: Using this `Cryptor` for encoding is strongly discouraged. Use ``AESCBCCryptor`` instead.
+public struct LegacyCryptor: Cryptor {
+  private let key: Data
+  private let withRandomIV: Bool
+  
+  static let ID: CryptorId = [0x00, 0x00, 0x00, 0x00]
+  
+  public init(key: String, withRandomIV: Bool = true) {
+    let hash = CryptorUtils.SHA256.hash(from: key.data(using: .utf8) ?? Data())
+    let hexStrData = CryptorUtils.hexFrom(hash).lowercased(with: .current).data(using: .utf8) ?? Data()
+    self.key = hexStrData
+    self.withRandomIV = withRandomIV
+  }
+  
+  public var id: CryptorId {
+    Self.ID
+  }
+  
+  public func encrypt(data: Data) -> Result<EncryptedData, Error> {
+    do {
+      let vectorGen = withRandomIV ? CryptorVector.random(bytesCount: kCCBlockSizeAES128) : CryptorVector.fixed
+      let ivData = try vectorGen.data()
+      
+      let encrypted = try data.crypt(
+        operation: CCOperation(kCCEncrypt),
+        algorithm: CCAlgorithm(kCCAlgorithmAES128),
+        options: CCOptions(kCCOptionPKCS7Padding),
+        blockSize: kCCBlockSizeAES128,
+        key: key,
+        initializationVector: ivData,
+        messageData: data
+      )
+      
+      // Join IV and encrypted content when using a random IV
+      return .success(EncryptedData(
+        metadata: ivData,
+        data: vectorGen.isRandom() ? ivData + encrypted : encrypted
+      ))
+    } catch {
+      return .failure(PubNubError(
+        .decryptionError,
+        underlying: error
+      ))
+    }
+  }
+  
+  public func decrypt(data: EncryptedData) -> Result<Data, Error> {
+    let iv: Data
+    let cipherText: Data
+    
+    do {
+      if withRandomIV {
+        iv = data.data.prefix(kCCBlockSizeAES128)
+        cipherText = data.data.suffix(from: kCCBlockSizeAES128)
+      } else {
+        iv = try CryptorVector.fixed.data()
+        cipherText = data.data
+      }
+      
```

Comment:
Should we have some more check here for case, when `cipherText` is empty after we extracted IV from it?